### DISCUSSION TITLE
classes/class_vector2i

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#223] [User-contributed notes]
# classes/class_vector2i

A 2D vector using integer coordinates. Description: A 2-element structure that can be used to represent 2D grid coordinates or any other pair of integers. It uses integer coordinates and is therefo...

https://docs.godotengine.org/en/stable/classes/class_vector2i.html

**Comment by berarma** [comment]
Vector2i(Vector2) truncating to zero may lead to unexpected results. Operations with floats have precision issues. While Godot might show a float value as an exact integer, internally it might be not.

After doing integer calculations with a Vector2 variable, truncating a supposedly integer value may give as a result a different integer value (value - 1 for positive values, value + 1 for negative values).

Example:
```
		print(Vector2(0, 0.9999995))
		print(Vector2i(Vector2(0, 0.9999995)))
```

Result:
```
(0, 1)
(0, 0)
```

**Reply by RayChimera** [reply]
Thank you, it looks like the bug I caught is directly connected to this issue

**Comment by paolomorettin** [comment]
Minor suggestion: even if this can be somehow inferred from the return type, I would explicitly state that distance_to returns the *Euclidean* distance (to me, returning the Manhattan distance when using integer vectors was reasonable too).

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/119222
